run_tests.py

```diff
@@ -80,24 +83,47 @@ class GRREverythingTestLoader(test_lib.GRRTestLoader):
   base_class = test_lib.GRRBaseTest
 
 
+class GRROneTestLoader(test_lib.GRRTestLoader):
+  """Load one specific GRR test case."""
+  base_class = test_lib.GRRBaseTest
+
+  def loadTestsFromModule(self, _):
+    """Just return all the tests as if they were in the same module."""
+    test_cases = [
+        self.loadTestsFromTestCase(x) for x in self.base_class.classes.values()
+        if x.__name__ == self.test_class_name]
+
+    return self.suiteClass(test_cases)
+
+  def __init__(self, test_class_name):
+    self.test_class_name = test_class_name
+
+    super(GRROneTestLoader, self).__init__()
+
+
 def RunTest(test_suite, stream=None):
   out_fd = stream
   if stream:
     out_fd = StringIO.StringIO()
 
   try:
-    test_lib.GrrTestProgram(argv=[sys.argv[0], test_suite],
-                            testLoader=GRREverythingTestLoader(
-                                labels=flags.FLAGS.labels),
-                            testRunner=unittest.TextTestRunner(
-                                stream=out_fd))
+    program = test_lib.GrrTestProgram(
+                argv=[sys.argv[0], test_suite],
+                testLoader=GRROneTestLoader(
+                    test_class_name=test_suite),
+                testRunner=unittest.TextTestRunner(
+                    stream=out_fd))
+  except SystemExit as e:
+    # must include this section, as GrrTestProgram issues
```

Comment:
The single test mode is designed for debugging. When you run the test suite and it tells you test XXYY is failing, then you just run the same with the --test XXYY and then it runs the test inline. You can then put break points which work because its running inline (if the subprocess is forked you cant interact with it since it has no terminal). I dont  think anyone really runs it on multiple test cases inline. It may be that it does not work when given multiple test cases due to the loop exiting on the first test - it is only ever run on one test at the time. If this is a bug lets fix it rather than working around a bug in a hacky way.

Multiprocess mode does work fine on Travis - we use it in Rekall without trouble. We use a similar test runner to the GRR one in Rekall (It might have been slightly updated, im not sure). Running each test suite in its own process is the right way to go rather than running all tests in the same process. Note that one of the nice things about running in multiprocess mode is that we can separate the output of the test into different files and then the runner can just present a green or red overview for a failed test. This output is much more useful then the output you show. Check this for example:
https://travis-ci.org/google/rekall/builds/49599483

In Rekall we also get OOM failures and we are thinking how to solve these. It happens rarely but when we have so many tests it means one random test might fail. We might re-run the test in the case of OOM at least a couple of times. This seems to be a limitation travis as a platform. 

I agree you may have discovered some bug, that's great. Lets fix them and move on.

If the purpose of the CL is to run the test suite on travis, I can not consider it complete until all the tests are passing on travis. I dont care too much if you run them in multiprocess or single process mode (this sounds to me like a limitation of the platform) and if you find bugs in the existing code that's great - lets fix those. The output you show indicates some tests are legitimately failing - we should fix them to ensure that they all pass.